pkg/ds/daemon_set.go

```diff
@@ -304,12 +313,26 @@ func (ds *daemonSet) WatchDesires(
 			case <-quitCh:
 				return
 			}
+
+			if err == nil {
+				err = reportErr
```

Comment:
argh no, this doesn't happen if we `continue`. This is why this is hard. I have to move this to the top of the function now. I don't want to put it after every `continue` since there are too many of them!